libreflect/src/stack_setup.c

```diff
@@ -96,16 +96,16 @@ void stack_setup(size_t *stack_base, int argc, char **argv, char **env, size_t *
 	dprint("  0x%08zx\n", stack_base[1 + ii]);
 
 	for (ii = 0; env[ii]; ii++) {
-		stack_base[1 + argc + ii] = (size_t)env[ii];
-		dprint("  0x%08zx\n", stack_base[1 + argc + ii]);
+		stack_base[2 + argc + ii] = (size_t)env[ii];
+		dprint("  0x%08zx\n", stack_base[2 + argc + ii]);
 	}
-	stack_base[1 + argc + ii] = 0;
-	dprint("  0x%08zx\n", stack_base[1 + argc + ii]);
+	stack_base[3 + argc + ii] = 0;
```

Comment:
Speaking of off-by-one errors this needs to be in line with the environment vector  ;). As above, to make that clear I would format it as:
```suggestion
	stack_base[1 + argc + ii +1] = 0;
```